### ISSUE TITLE
A .meta.yaml attribute to unlock the next map in the campaign without having to complete the map that has it

### ISSUE THREAD

**Issue by NeoKat75** [EverestAPI/Everest#221]
Currently every standalone map has to be put in a separate campaign to be playable without having to complete any other maps, causing some unnecessary clutter in the map list. I propose an Everest attribute for a map's .meta.yaml file that will unlock the next map in the campaign the map is in automatically (or do nothing if it's the last map in it). That way a creator could put all their standalone maps in one campaign, put that attribute in every map in it and they'll all be unlocked from the get go on any file. Perhaps assist skip could be used to achieve this?

**Comment by NeoKat75** [comment]
Also, what about C-Sides in 'optional' maps like this? Could it be unlocked after completing the B-Side of a map with this attribute enabled? Actually, I'm not sure if it's even possible to unlock only one C-Side...

**Comment by maddie480** [comment]
> Actually, I'm not sure if it's even possible to unlock only one C-Side...

No, you unlock all C-Sides at the same time. 😅 and afaik map meta.yamls are per-chapter, not per-side, so you cannot have a different one for a B-side anyway.

**Comment by 0x0ade** [comment]
> and afaik map meta.yamls are per-chapter, not per-side, so you cannot have a different one for a B-side anyway.

Mode properties and meta overriding *are* a thing though, and they're either stored in the .meta.yaml, A side meta or as part of the given side's "meta in bin".

**Comment by maddie480** [comment]
True, mode properties are a thing... so it _is_ possible, but I think that would be a separate attribute, since it does a completely different thing ("unlock the next chapter in the campaign directly" vs "only requiring you to beat the B-side before unlocking the C-side")... and not necessarily one associated to a side in particular, which makes my comment quite invalid. :sweat:

**Comment by coloursofnoise** [comment]
> Also, what about C-Sides in 'optional' maps like this? Could it be unlocked after completing the B-Side of a map with this attribute enabled? Actually, I'm not sure if it's even possible to unlock only one C-Side...

This is now possible through the [AltSidesHelper](https://github.com/l-Luna/AltSidesHelper/blob/master/DOCUMENTATION.md) mod

**Comment by microlith57** [comment]
resurrecting this; this would probably tie in pretty well with the proposed level visibility API ([discord link](https://discord.com/channels/403698615446536203/429775260108324865/1163088928061005876)), which really ought to have a meta.yaml attribute also
perhaps these could be implemented at the same time?
